stake-pool/program/src/processor.rs

```diff
@@ -316,112 +550,121 @@ impl Processor {
             withdraw_info.clone(),
             Self::AUTHORITY_WITHDRAW,
             stake_pool.withdraw_bump_seed,
-            user_amount,
+            token_amount,
         )?;
 
-        let fee_amount = <u64>::try_from(fee_amount).or(Err(Error::CalculationFailure))?;
-        Self::token_mint_to(
-            stake_pool_info.key,
-            token_program_info.clone(),
-            pool_mint_info.clone(),
-            owner_fee_info.clone(),
-            withdraw_info.clone(),
-            Self::AUTHORITY_WITHDRAW,
-            stake_pool.withdraw_bump_seed,
-            fee_amount as u64,
-        )?;
-        let pool_amount = <u64>::try_from(pool_amount).or(Err(Error::CalculationFailure))?;
-        stake_pool.pool_total += pool_amount;
+        // Add validator to the list and save
```

Comment:
Can you please send me the logic of how it can be done? Also I think this will mess up our tests (because stake accounts are always fresh ones there), how do you think we can deal with those?

Reply:
You can use the logic at https://github.com/solana-labs/solana/blob/master/programs/stake/src/stake_state.rs#L256 for that, and check that `activating` and `deactivating` are 0, and that `effective > 0`

Reply:
But it would still mess up the tests, right?

Reply:
It looks like 150 lines of code to be adopted from there for the activation logic (`stake_activating_and_deactivating` plus `stake_and_activating`), can we move it to the next PR?

Reply:
Sure, no worries, can you at least put a TODO in the code then?  And you can have it work for tests by putting the check under some `cfg(not(test))` flag or `cfg(not(feature = "test-bpf"))`

Reply:
Sure, TODO added :)